### PR TITLE
Samples Builder Image encodes runImage in label

### PR DESCRIPTION
**Pull request by matthewmcnew**
Signed-off-by: Andrew Meyer <ameyer@pivotal.io>
Signed-off-by: Matthew McNew <mmcnew@pivotal.io>

### CODE REVIEW

File: images/bin/build [hunk 1]
```diff
@@ -29,14 +29,18 @@ done
 
 cd $(dirname "${BASH_SOURCE[0]}")/..
 
-docker pull "${base}"
+docker pull $base
```
Comment (sclevine) [inline images/bin/build]: This should remain quoted, as it's never intended to be more than one argument.

File: images/bin/build [hunk 2]
```diff
@@ -29,14 +29,18 @@ done
...
 
-docker build --build-arg "base=${base}" -t "${prefix}base:${version}" ..
-docker build --build-arg "base=${prefix}base:${version}" -t "${prefix}build:${version}" build
-docker build --build-arg "base=${prefix}base:${version}" -t "${prefix}run:${version}" run
-docker build --build-arg "base=${prefix}build:${version}" -t "${prefix}samples:${version}" samples
+baseImage="${prefix}base:${version}"
```
Comment (sclevine) [inline images/bin/build]: RHS of assignment does not need quotes in bash.

File: images/bin/build [hunk 3]
```diff
@@ -29,14 +29,18 @@ done
...
+docker build --build-arg "base=${base}" -t $baseImage ..
+docker build --build-arg "base=${baseImage}" -t $buildImage build
+docker build --build-arg "base=${baseImage}" -t $runImage run
+docker build --build-arg "base=${buildImage}" --build-arg runImage="${runImage}"  -t $samplesBuilderImage samples
 
 echo "To publish these images:"
-for suffix in base build run samples; do
-  echo "  docker push \"${prefix}${suffix}:${version}\""
-done
+for image in $baseImage $runImage $buildImage $samplesBuilderImage; do
```
Comment (sclevine) [inline images/bin/build]: These need quotes to prevent 5+ iterations from happening.

File: images/bin/build [hunk 4]
```diff
@@ -29,14 +29,18 @@ done
...
+docker pull $base
 
-docker build --build-arg "base=${base}" -t "${prefix}base:${version}" ..
-docker build --build-arg "base=${prefix}base:${version}" -t "${prefix}build:${version}" build
-docker build --build-arg "base=${prefix}base:${version}" -t "${prefix}run:${version}" run
-docker build --build-arg "base=${prefix}build:${version}" -t "${prefix}samples:${version}" samples
+baseImage="${prefix}base:${version}"
+runImage="${prefix}run:${version}"
+buildImage="${prefix}build:${version}"
+samplesBuilderImage="${prefix}samples:${version}"
```
Comment (sclevine) [inline images/bin/build]: How about `samplesImage`?

File: images/bin/build [hunk 5]
```diff
@@ -29,14 +29,18 @@ done
...
+docker build --build-arg "base=${base}" -t $baseImage ..
```
Comment (sclevine) [inline images/bin/build]: Missing quotes for `$baseImage`, etc.

File: images/samples/Dockerfile [hunk 6]
```diff
@@ -18,4 +18,7 @@ RUN \
   done && \
   rm -rf /tmp/samples
 
+ARG runImage
```
Comment (sclevine) [inline images/samples/Dockerfile]: Args should use underscores, like bash vars: `run_image`

File: images/bin/build (hunk 1 above)
Comment (ameyer-pivotal) [reply images/bin/build]: ➕

File: images/bin/build [hunk 7]
```diff
@@ -31,12 +31,16 @@ cd $(dirname "${BASH_SOURCE[0]}")/..
...
-docker build --build-arg "base=${prefix}base:${version}" -t "${prefix}run:${version}" run
-docker build --build-arg "base=${prefix}build:${version}" -t "${prefix}samples:${version}" samples
+baseImage=${prefix}base:${version}
+runImage=${prefix}run:${version}
+buildImage=${prefix}build:${version}
+samplesImage=${prefix}samples:${version}
+docker build --build-arg "base=${base}" -t "${baseImage}" ..
+docker build --build-arg "base=${baseImage}" -t "${buildImage}" build
+docker build --build-arg "base=${baseImage}" -t "${runImage}" run
+docker build --build-arg "base=${buildImage}" --build-arg run_image="${runImage}"  -t "${samplesImage}" samples
```
Comment (sclevine) [inline images/bin/build]: `run_image="${runImage}"` -> `"run_image=${runImage}"` to match others

**Pull request merged**